### PR TITLE
Remove support for end-of-life Python 2.7

### LINKED ISSUE #1063: Python 2.7 EOL plans (Leaving feedback is encouraged)
In a little over a month, the PSF will end support for Python 2. @fcurella, what do you intend to do with this project concerning Python 2 EOL? Do you plan to retain support for a certain duration after the EOL date?

In any case, I can help with making everything Python 3 only, but I think `Faker` should have at least one last "semi-significant", if not significant, release that is Python 2 compatible for the poor souls who are stuck with Python 2. Your thoughts?

### PR DESCRIPTION
**Pull request by jdufresne**
Python 2.7 went end of life at the end of 2019. It is no longer
receiving updates including for security issues.

Removing Python 2.7 support reduces testing resources and removes the
need to maintain compatibility shims.

- Remove unnecessary dependencies six and ipaddress.
- Remove unnecessary test dependencies: mock, more-itertools.
- Unpin random2 test dependency.
- Run pyupgrade to cleanup syntax.

As the project no longer supports Python 2, the wheel should no longer
be designated as "universal" and so setup.cfg has been updated. The
wheel can continue to be built for Python 3 only.

Fixes #1063

### CODE REVIEW

**Comment by jdufresne** [comment]
Using pypinfo, here are the PyPI download statistics from the last month:

| python_version | percent | download_count |
| -------------- | ------: | -------------: |
| 3.7            |  35.27% |        495,756 |
| 3.6            |  32.43% |        455,857 |
| 2.7            |  21.16% |        297,466 |
| 3.5            |   5.84% |         82,066 |
| 3.8            |   4.74% |         66,671 |
| 3.4            |   0.56% |          7,923 |
| 3.9            |   0.00% |             20 |
| 2.6            |   0.00% |              5 |
| Total          |         |      1,405,764 |

File: tox.ini [hunk 1]
```diff
@@ -1,17 +1,13 @@
 [tox]
-envlist=py{27,35,36,37,38,py,py3},32bit,flake8,checkmanifest,isort
+envlist=py{35,36,37,38,py3},32bit,flake8,checkmanifest
```
Comment (fcurella) [inline tox.ini]: Could you add `isort` back in?

**Comment by jdufresne** [comment]
Thanks for catching that. Added back in.

**Comment by fcurella** [comment]
Nice Work! ⭐️

**Comment by jdufresne** [comment]
Thank you! :slightly_smiling_face:

**Pull request merged**